pymc3/distributions/bound.py

```diff
@@ -232,32 +132,53 @@ class Bound:
     .. code-block:: python
 
         with pm.Model():
-            NegativeNormal = pm.Bound(pm.Normal, upper=0.0)
-            par1 = NegativeNormal('par`', mu=0.0, sigma=1.0, initval=-0.5)
-            # you can use the Bound object multiple times to
-            # create multiple bounded random variables
-            par1_1 = NegativeNormal('par1_1', mu=-1.0, sigma=1.0, initval=-1.5)
-
-            # you can also define a Bound implicitly, while applying
-            # it to a random variable
-            par2 = pm.Bound(pm.Normal, lower=-1.0, upper=1.0)(
-                    'par2', mu=0.0, sigma=1.0, initval=1.0)
+            normal_dist = Normal.dist(mu=0.0, sigma=1.0, initval=-0.5)
+            negative_normal = pm.Bound(normal_dist, upper=0.0)
+
     """
 
-    def __init__(self, distribution, lower=None, upper=None):
-        if isinstance(lower, Real) and lower == -np.inf:
-            lower = None
-        if isinstance(upper, Real) and upper == np.inf:
-            upper = None
+    def __new__(cls, name, distribution, lower=None, upper=None, size=None, **kwargs):
```

Comment:
If we are in a `modelcontext` we should check here that the `distribution` is not in `model` (which would mean the user did not use `.dist()` and the original variable would be accounted on the model logp. 

In addition we can add the check that it is a TensorVariable, if not, the user is probably trying to work with the old API